onnx/defs/printer.cc

```diff
@@ -39,56 +39,108 @@ std::ostream& operator<<(std::ostream& os, const TypeProto_Tensor& tensortype) {
     if (tensortype.shape().dim_size() > 0)
       os << tensortype.shape();
   } else
-    os << "[...]";
+    os << "[]";
 
   return os;
 }
 
+std::ostream& operator<<(std::ostream& os, const TypeProto_Sequence& seqType) {
+  return os << "seq(" << seqType.elem_type() << ")";
+}
+
+std::ostream& operator<<(std::ostream& os, const TypeProto_Map& mapType) {
+  return os << "map(" << PrimitiveTypeNameMap::ToString(mapType.key_type()) << ", " << mapType.value_type() << ")";
+}
+
+std::ostream& operator<<(std::ostream& os, const TypeProto_Optional& optType) {
+  return os << "optional(" << optType.elem_type() << ")";
+}
+
+std::ostream& operator<<(std::ostream& os, const TypeProto_SparseTensor sparseType) {
+  os << "sparse_tensor(" << PrimitiveTypeNameMap::ToString(sparseType.elem_type());
+  if (sparseType.has_shape()) {
+    if (sparseType.shape().dim_size() > 0)
+      os << sparseType.shape();
+  } else
+    os << "[]";
+
+  return os << ")";
+}
+
 std::ostream& operator<<(std::ostream& os, const TypeProto& type) {
   if (type.has_tensor_type())
     os << type.tensor_type();
+  else if (type.has_sequence_type())
+    os << type.sequence_type();
+  else if (type.has_map_type())
+    os << type.map_type();
+  else if (type.has_optional_type())
+    os << type.optional_type();
+  else if (type.has_sparse_tensor_type())
+    os << type.sparse_tensor_type();
   return os;
 }
 
 std::ostream& operator<<(std::ostream& os, const TensorProto& tensor) {
   os << PrimitiveTypeNameMap::ToString(tensor.data_type());
   print(os, "[", ",", "]", tensor.dims());
 
-  // TODO: does not yet handle raw_data or FLOAT16 or externally stored data.
```

Comment:
Try to understand more context about it: you removed the comment here for FLOAT16, because current ParseData for raw_data does not support FLOAT16?